capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+        # excluded prefix is not contained within any less-specific prefix
+        # in the included set, we can elide it.  In other words, if the
+        # next-less-specific prefix is the implicit "default except",
+        # there is no need to configure the more specific "except".
+        exclude_result = []
+        for exclude_prefix in exclude:
+            for include_prefix in include_result:
+                if exclude_prefix.subnet_of(include_prefix):
+                    exclude_result.append(exclude_prefix)
+                    break
+
+        return include_result, exclude_result
+
+    def _GenPrefixFieldset(self, direction, name, pfxs, ex_pfxs, af):
+        field_list = ""
+
+        for p in pfxs:
+            field_list += (" " * 6) + "%s\n" % p
+        for p in ex_pfxs:
+            field_list += (" " * 6) + "except %s\n" % p
+
+        fieldset_hdr = (
+            "field-set " + af + " prefix " +
+            direction + "-" + ("%s" % name) + "\n"
+        )
+        field_set = fieldset_hdr + field_list
+        return field_set
+
+    def _TranslatePolicy(self, pol, exp_info):
+        self.arista_traffic_policies = []
+        _AF_MAP_TXT = {"inet": "ipv4", "inet6": "ipv6"}
+
+        current_date = datetime.datetime.utcnow().date()
+        exp_info_date = current_date + datetime.timedelta(weeks=exp_info)
+
+        for header, terms in pol.filters:
+            if self._PLATFORM not in header.platforms:
+                continue
+
+            filter_options = header.FilterOptions(self._PLATFORM)
+            filter_name = header.FilterName(self._PLATFORM)
+            noverbose = "noverbose" in filter_options[1:]
+
+            term_names = set()
+            new_terms = []           # list of generated terms
+            policy_field_sets = []   # list of generated field-sets
+            policy_counters = set()  # set of the counters in the policy
+
+            # default to mixed policies
+            filter_type = "mixed"
+            if len(filter_options) > 1:
+                filter_type = filter_options[1]
+
+            # if the filter_type is mixed, we need to iterate through the
+            # supported address families. treat the incoming policy term
+            # (pol_term) as a template for the term and override the necessary
+            # of the term for the inet6 evaluation.
+            #
+            # only make a copy of the pol_term if filter_type = "mixed"
+            ftypes = []
+            if filter_type == "mixed":
+                ftypes = ["inet", "inet6"]
+            else:
+                ftypes = [filter_type]
+
+            for pol_term in terms:
+                for ft in ftypes:
+                    if filter_type == "mixed":
+                        term = copy.deepcopy(pol_term)
+                    else:
+                        term = pol_term
+
+                    term_type = ft
+                    term.name = self.FixTermLength(term.name)
```

Comment:
Is there a limit to the term-name length specified by the platform? 

Currently this is set to the default of 62, after which FixTermLength throws an error.

Reply:
If you end up having to truncate term names, that could result in an issue as well. 

Corresponding requirement - https://github.com/google/capirca/blob/master/doc/generator_patterns.md#perform-truncating-of-names-for-term-name-or-comment-based-on-max-length